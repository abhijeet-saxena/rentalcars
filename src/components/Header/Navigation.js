import React from 'react';
import { NavContainer } from './styles';
import { Flex, Box, colors } from '../../common/components';
import logo from '../../assets/images/rc-logo.svg';
import gb from '../../assets/images/gb.png';

const Navigation = () => {
    return (
        <NavContainer>
            <a href="/">
                <img src={logo} alt="Rentalcars.com Logo"></img>
            </a>
            <Flex
                width="auto"
                gap="1.5rem"
                className="menu"
                alignItems="center"
            >
                <span style={{ color: colors.textLight }}>INR</span>
                <img src={gb} alt="gb"></img>
                <button>Manage Booking</button>
            </Flex>

            <Box className="hamburger">
                <svg viewBox="0 0 24 24" width="20">
                    <path
                        stroke="currentColor"
                        fill="currentColor"
                        d="M2.25 18.753h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5zm0-6h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5zm0-6h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5z"
                    />
                </svg>
            </Box>
        </NavContainer>
    );
};

export default Navigation;
