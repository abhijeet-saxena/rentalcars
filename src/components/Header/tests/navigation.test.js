import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Navigation from '../index';

afterEach(cleanup);

it('Rendered Snapshot Matches', () => {
    const navigation = render(<Navigation />);
    expect(navigation).toMatchSnapshot();
});
