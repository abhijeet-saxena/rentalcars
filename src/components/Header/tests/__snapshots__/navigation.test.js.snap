// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Rendered Snapshot Matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-iCoGMd emAaQC"
        data-testid="navigation"
      >
        <a
          href="/rentalcars"
        >
          <img
            alt="Rentalcars.com"
            height="21"
            src="rc-logo.svg"
            width="156"
          />
        </a>
        <div
          class="sc-bdnxRM gjXIaz menu"
          width="auto"
        >
          <span
            aria-label="Prices in Indian Rupees"
            style="color: rgb(255, 255, 255);"
          >
            INR
          </span>
          <img
            alt="Selected Language English"
            src="gb.png"
          />
          <button
            type="button"
          >
            Manage Booking
          </button>
        </div>
        <button
          aria-label="Open Menu"
          class="hamburger"
          type="button"
        >
          <img
            alt=""
            src="hamburger.svg"
          />
        </button>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-iCoGMd emAaQC"
      data-testid="navigation"
    >
      <a
        href="/rentalcars"
      >
        <img
          alt="Rentalcars.com"
          height="21"
          src="rc-logo.svg"
          width="156"
        />
      </a>
      <div
        class="sc-bdnxRM gjXIaz menu"
        width="auto"
      >
        <span
          aria-label="Prices in Indian Rupees"
          style="color: rgb(255, 255, 255);"
        >
          INR
        </span>
        <img
          alt="Selected Language English"
          src="gb.png"
        />
        <button
          type="button"
        >
          Manage Booking
        </button>
      </div>
      <button
        aria-label="Open Menu"
        class="hamburger"
        type="button"
      >
        <img
          alt=""
          src="hamburger.svg"
        />
      </button>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
